### ISSUE TITLE
Fragile Mind should be immune to N'loth's Gift event

### ISSUE THREAD

**Issue by dbjorge** [dbjorge/jorbs-spire-mod#57]

**Comment by wang429** [comment]
The class that handles this event is Nloth.class
```
        ArrayList<AbstractRelic> relics = new ArrayList();
        relics.addAll(AbstractDungeon.player.relics);
        Collections.shuffle(relics, new Random(AbstractDungeon.miscRng.randomLong()));
        this.choice1 = (AbstractRelic)relics.get(0);
        this.choice2 = (AbstractRelic)relics.get(1);
```

**Comment by wang429** [comment]
In the base game's EventHelper, `EventHelper.getEvent()` simply instantiates and returns a new instance of Nloth event class.

**Comment by dbjorge** [comment]
This would be a good bug for someone looking to try out making a `@SpirePatch`; I think a reasonable way to implement this would be to use a `SpireInsertPatch` the looks for the `Collections.shuffle` call in the `Nloth.class`'s `<ctor>`, and removes non-candidate relics from the `relics` local.

I think just hard-coding Fragile Mind would be acceptable for this bug fix, but even better would be to add some sort of Annotation or marker interface (maybe `NonRemovableRelic`) for `Fragile Mind` to use that the patch could look for.

**Comment by wang429** [comment]
Gonna work on this, seems important enough to get out of the way.

**Comment by wang429** [comment]
Need to also update `AbstractDungeon::getShrine` line 2239 to check the size not including Fragile Mind, otherwise, it's possible to crash with `java.lang.IndexOutOfBoundsException` when instantiating the event.

**Comment by dbjorge** [comment]
Fixed by #206
